experimental/Schemes/SpaceGerms.jl

```diff
@@ -225,53 +850,126 @@ function SpaceGerm(A::LocalRing)
   return SpaceGerm(Spec(A))
 end
 
+function HypersurfaceGerm(A::LocalRing)
+  I = modulus(A)
+  v  = minimal_generating_set(I)
+  length(v) == 1 || error("not a hypersurface germ")
+  return HypersurfaceGerm(Spec(A),v[1])
+end
+
+function CompleteIntersectionGerm(A::LocalRing)
+  I = modulus(A)
+  !iszero(I) || error("zero ideal not allowed for complete intersection germ")
+  v = minimal_generating_set(I)
+  length(v) == dim(base_ring(I)) - dim(A) || error("not a complete intersection germ")
+  return CompleteIntersectionGerm(Spec(A),v)
+end
+
 ## and with identity map to keep usage consistent
 function germ_at_point(A::LocalRing)
   X = SpaceGerm(A)
   restr_map = SpecMor(X, X, hom(OO(X), OO(X), gens(OO(X)), check=false), check=false)
   return X, restr_map
 end
 
+function hypersurface_germ(A::LocalRing)
+  X = HypersurfaceGerm(A)
+  restr_map = SpecMor(X, X, hom(OO(X), OO(X), gens(OO(X)), check=false), check=false)
+  return X, restr_map
+end
+
+function complete_intersection_germ(A::LocalRing)
+  X = CompleteIntersectionGerm(A)
+  restr_map = SpecMor(X, X, hom(OO(X), OO(X), gens(OO(X)), check=false), check=false)
+  return X, restr_map
+end
+
 ### basic functionality for space germs
 
 ##############################################################################
-# note: ==, intersect are inherited from Spec
-#       intersect with explicit fallback to Spec and change of return type
+# note: ==, issubset are basically inherited from Spec
+#       intersect uses explicit fallback to Spec and adjusted return types
+#       union uses explicit fallback and adjusted return types
 ##############################################################################
+@doc raw"""
+    is_subset(X::AbsSpaceGerm,Y::AbsSpaceGerm)
 
-function issubset(X::AbsSpaceGerm{<:Any, <:MPolyQuoLocRing}, Y::AbsSpaceGerm{<:Any, <:MPolyQuoLocRing})
+Returns whether `X` is a subset of `Y` as space germs
```

Comment:
We probably do not want this documentation of methods of such universal functions.

Reply:
I commented out the docstring for is_subset, but I left in intersect and union, because there the return types matter for the user and this should be mentioned in the documentation.